test/spec.controller.js

```diff
@@ -64,63 +163,210 @@ describe('Form Controller', function () {
+        let controller;
+
+        beforeEach(() => {
+            controller = new Controller(options);
+            sinon.stub(Form.prototype, 'use');
+        });
+
+        afterEach(() => {
+            Form.prototype.use.restore();
+        });
+
+        it('should call the parent use with a flattened list of middleware', () => {
+            let fn = sinon.stub();
+            controller.use([fn, [fn, fn, [fn, fn]]]);
+            Form.prototype.use.should.have.been.calledWithExactly(
+                sinon.match.func,
+                sinon.match.func,
+                sinon.match.func,
+                sinon.match.func,
+                sinon.match.func
+            );
+        });
+
+        it('should call use with functions bound to the controller instance', () => {
+            let fn = sinon.stub();
+            Form.prototype.use.yields();
+            controller.use(fn);
+            fn.should.have.been.calledOn(controller);
+        });
+    });
+
+
+    describe('requestHandler', () => {
+        let controller;
+
+        beforeEach(() => {
+            controller = new Controller(options);
+            sinon.stub(Form.prototype, 'requestHandler');
+            sinon.stub(Controller.prototype, 'middlewareSetup');
+            sinon.stub(Controller.prototype, 'middlewareChecks');
+            sinon.stub(Controller.prototype, 'middlewareActions');
+            sinon.stub(Controller.prototype, 'middlewareLocals');
+        });
+
+        afterEach(() => {
+            Form.prototype.requestHandler.restore();
+            Controller.prototype.middlewareSetup.restore();
+            Controller.prototype.middlewareChecks.restore();
+            Controller.prototype.middlewareActions.restore();
+            Controller.prototype.middlewareLocals.restore();
+        });
+
+        it('should call the middleware setup methods', () => {
+            controller.requestHandler();
+            Controller.prototype.middlewareSetup.should.have.been.calledOnce;
+            Controller.prototype.middlewareChecks.should.have.been.calledOnce;
+            Controller.prototype.middlewareActions.should.have.been.calledOnce;
+            Controller.prototype.middlewareLocals.should.have.been.calledOnce;
+        });
+
+        it('should call form-controllers requestHandler method', () => {
+            controller.requestHandler();
+            Form.prototype.requestHandler.should.have.been.calledOnce;
+        });
+
+        it('should return the router from the form-controllers requestHandler', () => {
+            Form.prototype.requestHandler.returns('A Router');
+            controller.requestHandler().should.equal('A Router');
+        });
+
+    });
+
+    describe('middleware setup functions', () => {
+        it('should call be called by the request handler', () => {
+            let controller = new Controller(options);
+            controller.requestHandler();
+            // TODO: prove that the middleware functions are being called
```

Comment:
This has been changed to running each setup method and checking that `use` has been called